genome_designer/main/signals.py

```diff
@@ -73,6 +73,8 @@ def pre_project_delete(sender, instance, **kwargs):
 def post_contig_delete(sender, instance, **kwargs):
     vccd = instance.variant_caller_common_data
     if vccd:
+        if vccd.variant.variantcallercommondata_set.count() == 1:
+            vccd.variant.delete()
         vccd.delete()
```

Comment:
vccd will be deleted if vccd.variant is deleted so more correct to do:

    if vccd.variant.variantcallercommondata_set.count() == 1:
         vccd.variant.delete()
        # vccd also deleted automatically
    else:
        vccd.delete()